kettle-plugins/hdfs/src/main/java/org/pentaho/big/data/kettle/plugins/hdfs/job/JobEntryHadoopCopyFiles.java

```diff
@@ -69,15 +86,36 @@ public String loadURL( String url, String ncName, IMetaStore metastore, Map mapp
       pref = origUrl.substring( 0, origUrl.indexOf( "-", origUrl.indexOf( DEST_URL ) + DEST_URL.length() ) + 1 );
     }
     if ( c != null ) {
+      String valueBeforeCall = url;
       url = c.processURLsubstitution( url, metastore, getVariables() );
+      saveArgumentUrl = !Objects.equals( valueBeforeCall, url );
     }
     if ( pref != null ) {
       url = pref + url;
     }
-    if ( !Const.isEmpty( ncName ) && !Const.isEmpty( url ) ) {
```

Comment:
same logic as parent class and should always calls super function, see below
`super.loadURL(url...`